### PR TITLE
Combine custom logger global threadpools

### PR DESCRIPTION
**Pull request by jgbradley1**
**Description**:

This PR resolves issue https://github.com/microsoft/onnxruntime/issues/4843

Changes are made to the C and C++ APIs to allow C++ users the ability to use both a global threadpool and custom logger.

I'm open to changing any function names to better conform to the API if there are any concerns.

### CODE REVIEW

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 1]
```diff
@@ -990,6 +988,16 @@ struct OrtApi {
...
   ORT_API2_STATUS(TensorAt, _Inout_ OrtValue* value, size_t* location_values, size_t location_values_count, _Outptr_ void** out);
+
+  /*
+  * Creates a custom environment with global threadpools and logger that will be shared across sessions.
+  * Use this in conjunction with DisablePerSessionThreads API or else the session will use
+  * its own thread pools.
+  * 
+  * \param out should be freed by `OrtReleaseEnv` after use
+  */
+  ORT_API2_STATUS(CreateCustomEnv, OrtLoggingFunction logging_function, _In_opt_ void* logger_param, OrtLoggingLevel default_logging_level,
```
Comment (pranavsharma) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:999]: I think adding a SetCustomLogger(_In_ const OrtEnv* env) function might be better. It gives us the flexibility to add more CreateEnv* variants later without creating the burden to accept a custom logger in those variants.

Comment (jgbradley1) [reply include/onnxruntime/core/session/onnxruntime_c_api.h:1022]: Not sure I understand. The argument variants already exist - they unfortunately did not allow for the combination of **all** arguments at the same time. A `SetCustomLogger(...)` function would be nice but should that be done in another PR?

Although verbose, I could rename the function to `CreateEnvWithCustomLoggerAndGlobalThreadPools(...)` to coincide with other functions naming style.

Comment (jgbradley1) [reply include/onnxruntime/core/session/onnxruntime_c_api.h]: @pranavsharma Sorry for the delay (been on vacation). Renamed the function as we discussed before and updated the branch from master. Should be good to go.

**Comment by pranavsharma** [comment]
@jgbradley1 while you were on vacation, we had to cut a branch for 1.5 release. Hence the API version is now 6 in the master. If you need this API, please add it in the version 6 section below AddInitializer. Thanks!

**Comment by jgbradley1** [comment]
@pranavsharma Updated the code so it's part of version 6 now.

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**